src/Core/Upload/ResumableUploader.php

```diff
@@ -121,6 +157,11 @@ public function upload()
                     $ex->getCode()
                 );
             }
+            
+            if (is_callable($this->uploadProgressCallback))
+            {
+                call_user_func($this->uploadProgressCallback, $rangeEnd - $rangeStart + 1);
```

Comment:
Instead of `$rangeEnd - $rangeStart + 1` we could just get the size of the `LimitStream` above. You could cache the value into a variable to be used here and when setting the `Content-Length`.